src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -77,41 +91,295 @@
+            BreedingBlockComponent breedingBlockComponent = blockEntity.getComponent(BreedingBlockComponent.class);
+
+            if (followComponent.entityToFollow != EntityRef.NULL && breedingBlockComponent.parentGooey == EntityRef.NULL && gooeyComponent.isCaptured) {
+                breedingBlockComponent.parentGooey = entity;
+                entity.send(new SelectForBreedingEvent(followComponent.entityToFollow, entity));
+                blockEntity.saveComponent(breedingBlockComponent);
+            }
+        }
+    }
+
+    /**
+     * Receives the SelectForBreedingEvent when the "activated" gooey is chosen for breeding.
      *
      * @param event
      * @param gooeyEntity
      * @param gooeyComponent
      */
     @ReceiveEvent
-    public void onBreedingGooey(BreedGooeyEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent, HungerComponent hungerComponent) {
-        CharacterHeldItemComponent characterHeldItemComponent = event.getInstigator().getComponent(CharacterHeldItemComponent.class);
-
-        if (characterHeldItemComponent != null && characterHeldItemComponent.selectedItem.getComponent(DisplayNameComponent.class) != null) {
-            EntityRef item = characterHeldItemComponent.selectedItem;
-            String itemName = item.getComponent(DisplayNameComponent.class).name;
-
-            if (!itemName.isEmpty() && hungerComponent.food.contains(itemName)) {
-                logger.info("Selected for breeding ...");
-                FollowComponent followComponent = gooeyEntity.getComponent(FollowComponent.class);
-                BehaviorComponent behaviorComponent = gooeyEntity.getComponent(BehaviorComponent.class);
-                MatingComponent matingComponent = new MatingComponent();
-
-                followComponent.entityToFollow = event.getInstigator();
-                behaviorComponent.tree = assetManager.getAsset("GooKeeper:breedingBehavior", BehaviorTree.class).get();
-                CharacterMovementComponent characterMovementComponent = gooeyEntity.getComponent(CharacterMovementComponent.class);
-                characterMovementComponent.jumpSpeed = 12f;
-
-                gooeyEntity.saveComponent(followComponent);
-                gooeyEntity.saveComponent(behaviorComponent);
-                gooeyEntity.addOrSaveComponent(matingComponent);
-                gooeyEntity.saveComponent(characterMovementComponent);
+    public void onBreedingGooey(SelectForBreedingEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent) {
+        logger.info("Selected for breeding...");
+
+        MatingComponent matingComponent;
+
+        if (!gooeyEntity.hasComponent(MatingComponent.class)) {
+            matingComponent = new MatingComponent();
+        } else {
+            matingComponent = gooeyEntity.getComponent(MatingComponent.class);
+        }
+
+        matingComponent.selectedForMating = !matingComponent.selectedForMating;
+
+        if (matingComponent.selectedForMating) {
```

Comment:
This is a fairly big `if`.
 It may be worth moving the contents of it to a separate method, named appropriately